linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/java/org/apache/linkis/manager/rm/external/yarn/YarnResourceRequester.java

```diff
@@ -0,0 +1,426 @@
+
+    return LinkisUtils.tryCatch(
+        () -> {
+          Pair<YarnResource, YarnResource> yarnResource =
+              getResources(rmWebAddress, realQueueName, queueName);
+          getResources(rmWebAddress, realQueueName, queueName);
+
+          CommonNodeResource nodeResource = new CommonNodeResource();
+          nodeResource.setMaxResource(yarnResource.getKey());
+          nodeResource.setUsedResource(yarnResource.getValue());
+          return nodeResource;
+        },
+        t -> {
+          throw new RuntimeException(t);
+        });
+  }
+
+  public Optional<YarnResource> maxEffectiveHandle(
+      Optional<JsonNode> queueValue, String rmWebAddress, String queueName) {
+    try {
+      JsonNode metrics = getResponseByUrl("metrics", rmWebAddress);
+      long totalMemory = metrics.path("clusterMetrics").path("totalMB").asLong();
+      long totalCores = metrics.path("clusterMetrics").path("totalVirtualCores").asLong();
+      if (queueValue.isPresent()) {
+        JsonNode jsonNode = queueValue.get();
+        double absoluteCapacity = jsonNode.path("absoluteCapacity").asDouble();
+        double effectiveResource = absoluteCapacity;
+
+        YarnResource yarnResource =
+            new YarnResource(
+                (long) Math.floor(effectiveResource * totalMemory * 1024L * 1024L / 100),
+                (int) Math.floor(effectiveResource * totalCores / 100),
+                0,
+                queueName,
+                "");
+        return Optional.of(yarnResource);
+      }
+      return Optional.empty();
+    } catch (Exception e) {
+      // e.printStackTrace();
+      return Optional.empty();
+    }
+  }
+
+  public static Optional getQueue(JsonNode queues, String realQueueName) {
+    if (queues instanceof ArrayNode) {
+      for (JsonNode q : (ArrayNode) queues) {
+        String yarnQueueName = q.get("queueName").asText();
+        if (yarnQueueName.equals(realQueueName)) {
+          return Optional.of(q);
+        } else if (realQueueName.startsWith(yarnQueueName + ".")) {
+          Optional childQueue = getQueue(getChildQueues(q), realQueueName);
+          if (childQueue.isPresent()) {
+            return childQueue;
+          }
+        }
+      }
+      return Optional.empty();
+    } else if (queues instanceof ObjectNode) {
+      ObjectNode queueObj = (ObjectNode) queues;
+      JsonNode queueName = queueObj.get("queueName");
+      if (queueName != null && queueName.asText().equals(realQueueName)) {
+        return Optional.of(queueObj);
+      } else {
+        JsonNode childQueues = queueObj.get("childQueues");
+        if (childQueues == null) {
+          return Optional.empty();
+        } else {
+          return getQueue(childQueues, realQueueName);
+        }
+      }
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  public static JsonNode getChildQueues(JsonNode resp) {
+    JsonNode queues = resp.get("childQueues").get("queue");
```

Comment:
is `queue` sub-attribute of `childQueues ` ?

Reply:
this method is translate from pre scala version like bellow:
![image](https://user-images.githubusercontent.com/16526279/236689883-887cbc69-0fc1-4174-83ce-88b0c9744d42.png)

Reply:
you can double confirm with actual json values

Reply:
already modified getChildQueues method，thanks.